### PR TITLE
Add zindex for ws_surface

### PR DESCRIPTION
**Pull request by matthiasbeyer**
This PR targets #620.

I'm not quite sure whether this is the appropriate approach. Also,
please note that I set the default z index to 1000, which clearly
must be discussed.

### CODE REVIEW

**Comment by neithernut** [comment]
If the z-index were an int, we could have a _much_ intuitive default of `0`.

**Comment by matthiasbeyer** [comment]
So shall I dynamically return either a bool (false for a "something did not work") or an int on `get` here?

**Comment by matthiasbeyer** [comment]
... because I don't think we should entirely fail here if this does not work...

**Comment by neithernut** [comment]
This should solve automatically when porting it to the generalized attribute modification and query infrastructure.

**Comment by matthiasbeyer** [comment]
I will rewrite this now with #626 in mind.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
Would you mind _not_ removing that blank line from `abstaract_shell_surface.c` in your one commit? You're not doing anything in that file anymore, anyways.

**Comment by TheNeikos** [comment]
He will, the Z-index is going to be on the abstract shell, the z index of surfaces is automatical through their parent/child relationship.

**Comment by matthiasbeyer** [comment]
@TheNeikos So I should move the z index to the abstract shell type?

**Comment by neithernut** [comment]
Mh... yep, I didn't consider doing it this way, but if the position is made an offset relative to it's parent (the shell surface), so would the z-index. Since an offset doesn't make sense for the z-index, it would be an attribute of the shell surface only.
TL;DR: Yep, make it an attribute of `ws_abstract_shell_surface`.

**Comment by neithernut** [comment]
ACK for squashing.

**Comment by matthiasbeyer** [comment]
Done.

**Comment by neithernut** [comment]
ACK.
